src/CanvasRenderingContext2d.cc

```diff
@@ -1178,13 +1180,25 @@ NAN_METHOD(Context2d::DrawImage) {
   // Scale src
   float fx = (float) dw / sw;
   float fy = (float) dh / sh;
+  bool needScale = dw != sw || dh != sh;
+  bool needCut = sw != fw || sh != fh;
 
-  if (dw != sw || dh != sh) {
-    cairo_scale(ctx, fx, fy);
-    dx /= fx;
-    dy /= fy;
-    dw /= fx;
-    dh /= fy;
+  bool sameCanvas = surface == context->canvas()->surface();
+  bool needsExtraSurface = sameCanvas || needCut || needScale;
+  cairo_surface_t *surfTemp = NULL;
+  cairo_t *ctxTemp = NULL;
+
+  if (needsExtraSurface) {
+    surfTemp = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, dw, dh);
+    ctxTemp = cairo_create(surfTemp);
+    cairo_save(ctxTemp);
```

Comment:
I don't think you need this `cairo_save` and the `cairo_restore` 6 lines down, do you?

Reply:
Yes was a left over from another change, removed.